generate/generate.go

```diff
@@ -244,14 +244,9 @@ func (g *Generator) ClearAnnotations() {
 }
 
 // AddAnnotation adds an annotation into g.spec.Annotations.
-func (g *Generator) AddAnnotation(s string) error {
+func (g *Generator) AddAnnotation(key, value string) error {
 	g.initSpecAnnotations()
-
-	pair := strings.Split(s, "=")
-	if len(pair) != 2 {
-		return fmt.Errorf("incorrectly specified annotation: %s", s)
-	}
-	g.spec.Annotations[pair[0]] = pair[1]
+	g.spec.Annotations[key] = value
 	return nil
```

Comment:
With this pattern, there are now a whole bunch of methods which call `g.init…()` and then do some simple struct work.  I still think the API would be easier to use and maintain if we made the `init…` helpers and `g.Config` public and dropped the thin wrappers around struct manipulation.